driver/src/price_finding/naive_solver.rs

```diff
@@ -115,48 +115,57 @@ impl PriceFinding for NaiveSolver {
     fn find_prices(&self, orders: &[Order], state: &AccountState) -> Result<Solution> {
         // Initialize trivial solution (default of zero indicates untouched token).
         let mut prices = HashMap::new();
-        let mut exec_buy_amount: Vec<u128> = vec![0; orders.len()];
-        let mut exec_sell_amount: Vec<u128> = vec![0; orders.len()];
-
+        let mut executed_orders: Option<[ExecutedOrder; 2]> = None;
         let mut matching_orders_indices: Option<[usize; 2]> = None;
         'outer: for (i, x) in orders.iter().enumerate() {
-            for j in i + 1..orders.len() {
+            for (j, y) in orders.iter().enumerate().skip(i + 1) {
+                let mut set_matched_orders = |sell_amounts: [u128; 2], buy_amounts: [u128; 2]| {
```

Comment:
Can we outline this lambda please into a method. This is a bit hard to read.

Reply:
Not easily possible because it captures a lot from the environment which would have to be made parameters. I made it like this in order to introduce the least amount of changes to naive_solver. I'll do a cleanup PR of naive_solver after this one.